### ISSUE TITLE
Missing ssh files are reported incorrectly

### ISSUE THREAD

**Issue by JeffreyDevloo** [openvstorage/framework#802]
# Problem description

Received the following message when only /opt/OpenvStorage/.ssh/ files were missing.

```
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+++  An unexpected error occurred:                                                 +++
+++  Missing files:                                                                +++
+++  - Could not find file /opt/OpenvStorage/.ssh/id_rsa.pub on node with IP       +++
+++  10.100.199.151                                                                +++
+++  - Could not find file /opt/OpenvStorage/.ssh/known_hosts on node with IP      +++
+++  10.100.199.151                                                                +++
+++   - Could not find file /root/.ssh/id_rsa.pub on node with IP 10.100.199.151   +++
+++   - Could not find file /root/.ssh/known_hosts on node with IP 10.100.199.151  +++
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
```
# Possible root of the problem

The check is only happening on one file instead of the file in the loop.
Framework.ovs.lib.setup.py

```
for required_file in [known_hosts_ovs, known_hosts_root, ssh_public_key_ovs, ssh_public_key_root]:
 if not local_client.file_exists(ssh_public_key_ovs):
  missing_files.add('Could not find file {0} on node with IP {1}'.format(required_file, local_client.ip))
 if missing_files:
  raise ValueError('Missing files:\n - {0}'.format('\n - '.join(sorted(list(missing_files)))))
```
# Possible solution

Change the if condition:

```
 if not local_client.file_exists(required_file):
```

**Comment by khenderick** [comment]
To be added here: the actual cause of the message was that the `ovs` user was created with a wrong home directory, that caused the package installation scripts to not created the key pairs in the expected location.

However, this problem will cause that the setup doesn't validate other files (e.g. root's keypair).

**Comment by wimpers** [comment]
@khenderick how can it be that the ovs user was created with the wrong home directory. Is it a bug in the code or should we tackle this with documentation?

**Comment by wimpers** [comment]
nevermind, the ovs user was created before (waffle didn't show the referenced issue).

**Comment by khenderick** [comment]
@wimpers, In this case @JeffreyDevloo just installed a fresh Ubuntu and when the Ubuntu setup asks for a non-privileged user, Jeffrey just entered "ovs" as it's username. The Ubuntu installation then creates a user with that username which will gets it home directory created under /home/ovs.

I remember that back in the days the documentation mentioned this since back then the Ubuntu installation itself was also documented.

**Comment by khenderick** [comment]
@wimpers, there's still an issue in the code reported here that must be fixed.

**Comment by wimpers** [comment]
Set to won't fix as if the ovs user isn't previously created, you don't run into the issue.

**Comment by wimpers** [comment]
Not having an ovs user before starting the install will be covered by documentation.

**Comment by wimpers** [comment]
Re-opening as fixed so put into state verification for QA.

**Comment by dejonghb** [comment]
If there's already an (unexpected) ovs user on the system (with an "incorrect home directory" -- whatever that may be); shouldn't the setup warn about it?

**Comment by kinvaris** [comment]
@dejonghb makes a very good point, @wimpers.

**Comment by khenderick** [comment]
@dejonghb, @kinvaris, it basically does this already by complaining the expected files could not be found. But maybe the message could be enhanced a bit more.

**Comment by JeffreyDevloo** [comment]
Reproduction of the issue gave me the expected output this time:

### Steps
- Installed a fresh Ubuntu 14.04
- Configured a ovs user
- ran ovs setup

### Output

```
Exchanging SSH keys and updating hosts files


++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+++  An unexpected error occurred:                                             +++
+++  Missing files:                                                            +++
+++  - Could not find file /opt/OpenvStorage/.ssh/id_rsa.pub on node with IP   +++
+++  10.100.199.153                                                            +++
+++  - Could not find file /opt/OpenvStorage/.ssh/known_hosts on node with IP  +++
+++  10.100.199.153                                                            +++
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
```

### Test result

Test passed.
The missing files are now reported correctly.
Closing

### Package information
- openvstorage                       2.7.2-rev.3859.42b3488-1   amd64        openvStorage
- openvstorage-backend               1.7.2-rev.675.37ca5b8-1    amd64        openvStorage Backend plugin
- openvstorage-backend-core          1.7.2-rev.675.37ca5b8-1    amd64        openvStorage Backend plugin core
- openvstorage-backend-webapps       1.7.2-rev.675.37ca5b8-1    amd64        openvStorage Backend plugin Web Applications
- openvstorage-cinder-plugin         1.2.2-rev.32.948a8c1-1 amd64        OpenvStorage Cinder plugin for OpenStack
- openvstorage-core                  2.7.2-rev.3859.42b3488-1   amd64        openvStorage core 
- openvstorage-hc                    1.7.2-rev.675.37ca5b8-1    amd64        openvStorage Backend plugin HyperConverged
- openvstorage-health-check          2.0.0-rev.117.3212ea9-1    amd64        Open vStorage HealthCheck
- openvstorage-sdm                   1.6.2-rev.330.f06c8de-1    amd64        Open vStorage Backend ASD Manager
- openvstorage-test                  2.7.2-rev.980.0dbc80f-1    amd64        openvStorage autotest suite
- openvstorage-webapps               2.7.2-rev.3859.42b3488-1   amd64        openvStorage Web Applications
